app/src/main/java/nerd/tuxmobil/fahrplan/congress/utils/ConfirmationDialog.kt

```diff
@@ -0,0 +1,77 @@
+package nerd.tuxmobil.fahrplan.congress.utils
+
+import android.app.Dialog
+import android.content.Context
+import android.content.DialogInterface
+import android.os.Bundle
+
+import androidx.annotation.CallSuper
+import androidx.annotation.MainThread
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import nerd.tuxmobil.fahrplan.congress.R
+
+class ConfirmationDialog : DialogFragment() {
+
+    interface OnConfirmationDialogClicked {
+        fun onAccepted(dlgRequestCode: Int)
+    }
+
+    companion object {
+        const val BUNDLE_DLG_TITLE = "ConfirmationDialog.DLG_TITLE"
+        const val BUNDLE_DLG_REQUEST_CODE = "ConfirmationDialog.DLG_REQUEST_CODE"
+        const val FRAGMENT_TAG = "ConfirmationDialog.FRAGMENT_TAG"
+
+        fun newInstance(@StringRes dlgTitle: Int, requestCode: Int): ConfirmationDialog {
+            val dialog = ConfirmationDialog()
+
+            dialog.listener = null
+
+            val args = Bundle()
+            args.putInt(BUNDLE_DLG_TITLE, dlgTitle)
+            args.putInt(BUNDLE_DLG_REQUEST_CODE, requestCode)
+            dialog.arguments = args
+
+            dialog.isCancelable = false
+
+            return dialog
+        }
+    }
+
+    private var dlgTitle = 0
+    private var dlgRequestCode = 0
+    private var listener: OnConfirmationDialogClicked? = null
+
+    @MainThread
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val args = requireArguments()
+        dlgTitle = args.getInt(BUNDLE_DLG_TITLE)
+        dlgRequestCode = args.getInt(BUNDLE_DLG_REQUEST_CODE)
+    }
+
+    @MainThread
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val builder = AlertDialog.Builder(requireContext())
+            .setTitle(dlgTitle)
+            .setPositiveButton(R.string.dlg_delete_all_favorites_delete_all) { dialog: DialogInterface?, which: Int ->
+                if (listener != null) {
+                    listener!!.onAccepted(dlgRequestCode)
+                }
+            }
+            .setNegativeButton(android.R.string.cancel) { dialog: DialogInterface?, which: Int -> {
+                // Do nothing.
+            }}
+        return builder.create()
+    }
+
+    @MainThread
+    @CallSuper
+    override fun onAttach(context: Context) {
```

Comment:
Here, I would move the `onAttach()` function above `onCreate()` because then it follows the [fragment lifecycle](https://i.stack.imgur.com/1EEke.png).